### PR TITLE
[OAP-1924][OAP-CACHE]Decouple hearbeat message and use conf to determine whether to report locailty information

### PR DESCRIPTION
**Pull request by Aaxida**
We are using external DB to store cache locality info, but the old OAP report mechanism still reports cache locality to the driver through RPC. Need to decouple this.
 
1:Using external DB => not report cache locality to the driver through RPC. But still, need to report cache metrics to the driver through RPC.
2:Using report mechanism =>follow current behavior.

In the code, it will check the configuration of whether to use Redis.
If used, just report metrics information.
If not, report both metrics and locality information.

### CODE REVIEW

**Comment by Aaxida** [comment]
@offthewall123  Could you please help to review this?

File: oap-cache/oap/src/main/scala/org/apache/spark/sql/oap/rpc/OapRpcManagerSlave.scala [hunk 1]
```diff
@@ -51,11 +51,17 @@ private[spark] class OapRpcManagerSlave(
...
   protected def heartbeatMessages: Array[() => Heartbeat] = {
-    Array(
-      () => FiberCacheHeartbeat(
-        executorId, blockManager.blockManagerId, fiberCacheManager.status()),
-      () => FiberCacheMetricsHeartbeat(executorId, blockManager.blockManagerId,
-        CacheStats.status(fiberCacheManager.cacheStats, conf)))
+    if (SparkEnv.get.conf.get(OapConf.OAP_EXTERNAL_CACHE_METADB_ENABLED)) {
+      return Array(
+        () => FiberCacheMetricsHeartbeat(executorId, blockManager.blockManagerId,
+          CacheStats.status(fiberCacheManager.cacheStats, conf)))
```
Comment (offthewall123) [inline oap-cache/oap/src/main/scala/org/apache/spark/sql/oap/rpc/OapRpcManagerSlave.scala:57]: Did you have a E2E test to verify it?

**Comment by offthewall123** [comment]
OapFileSourceScanExec should also be modified. @Aaxida

**Comment by offthewall123** [comment]
LGTM. @jikunshang

**Comment by jikunshang** [comment]
LGTM. Merged. 
By the way, branch-1.0 also need this PR, right? If so, I will do cherry-pick .

**Comment by offthewall123** [comment]
> LGTM. Merged.
> By the way, branch-1.0 also need this PR, right? If so, I will do cherry-pick .

yes

**Comment by jikunshang** [comment]
I tried to cherry-pick directly, but there are some conflicts.
Can you help port this PR to branch-1.0? You can submit another PR and link to the same issue(1924) @Aaxida

**Pull request merged**